src/include/sof/spinlock.h

```diff
@@ -189,10 +189,10 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 		arch_spin_lock(lock); \
 	} while (0)
 
-#define spin_try_lock(lock) \
+#define spin_try_lock(lock, ret) \
 	do { \
 		spin_lock_dbg(); \
-		arch_try_lock(lock); \
+		ret = arch_try_lock(lock); \
```

Comment:
Wouldn't a better approach be to use a block expression? (GNU extension). As in,

```c
#define spin_try_lock(lock) ({ \
              spin_lock_dbg();  \
              arch_try_lock(lock); \
}
```

(with the proper alignment)? That would also be of lower impact to the remainder of the code.

Reply:
For consistency I think that we can change it for every spin_lock*() macro in another PR.

Reply:
ok, we could make this an incremental update in another PR.